cinema/views.py

```diff
@@ -1 +1,64 @@
-# write views here
+from rest_framework import viewsets
+from django.db.models import QuerySet
+
+from cinema.models import Movie, MovieSession, CinemaHall, Actor, Genre
+from cinema.serializers import (
+    MovieSerializer,
+    MovieListSerializer,
+    MovieRetireveSerializer,
+    MovieSessionSerializer,
+    MovieSessionListSerializer,
+    MovieSessionRetireveSerializer,
+    CinemaHallSerializer,
+    ActorSerializer,
+    GenreSerializer,
+)
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+
+    def get_serializer_class(self) -> MovieSerializer:
+        if self.action == "list":
+            return MovieListSerializer
+        if self.action == "retrieve":
+            return MovieRetireveSerializer
+        return MovieSerializer
+
+    def get_queryset(self) -> QuerySet:
+        queryset = self.queryset
```

Comment:
Consider removing the `.all()` method before `.prefetch_related()` and `.select_related()` methods. It is not necessary and can be omitted. The correct way to write this would be `queryset = Movie.objects.prefetch_related('actors', 'genres')`.